talktorials/11_online_apis/11c_quality_assessment.ipynb

```diff
@@ -0,0 +1,332 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def split_pdbqt(path):\n",
+    "    \"\"\"\n",
+    "    Split a multimodel PDBQT into separate files.\n",
+    "    \"\"\"\n",
+    "    files = []\n",
+    "    with open(path) as f:\n",
+    "        lines = []\n",
+    "        i = 0\n",
+    "        for line in f:\n",
+    "            lines.append(line)\n",
+    "            if line.strip() == 'ENDMDL':\n",
+    "                fn = f'data/results.{i}.pdbqt'\n",
+    "                with open(fn, 'w') as o:\n",
+    "                    o.write(''.join(lines))\n",
+    "                files.append(fn)\n",
+    "                i += 1\n",
+    "                lines = []\n",
+    "    return files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Vina output is a simple text file that contains the table of results. Parsing that table is relatively straightforward. We return a Pandas DataFrame for a simple visualization, if needed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def parse_output(out):\n",
+    "    \"\"\"\n",
+    "    Create a DataFrame out of the Vina output file\n",
+    "    \"\"\"\n",
+    "    with open(out) as f:\n",
+    "        data = []\n",
+    "        for line in f:\n",
+    "            if line.startswith('-----+'):\n",
+    "                line = next(f)\n",
+    "                while line.split()[0].isdigit():\n",
+    "                    index, *floats = line.split()\n",
+    "                    data.append([int(index)] + list(map(float, floats)))\n",
+    "                    line = next(f)\n",
+    "    return pd.DataFrame.from_records(data, \n",
+    "                                     columns=['Mode', 'Affinity (kcal/mol)', 'RMSD (l.b.)', 'RMSD (u.b.)'], \n",
+    "                                     exclude=['Mode'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we create the NGL viewer instance. Instead of creating a new one for each protein-pose pair, we will reuse the same canvas all over, hiding or showing the needed ligands. We will load everything first, while also labeling the ligands with their respective affinity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_viewer(protein, ligands, affinities):\n",
+    "    \"\"\"\n",
+    "    Create a nglview widget with the protein and all the ligands labeled by affinities\n",
+    "    \"\"\"\n",
+    "    v = nv.show_file(protein)\n",
+    "    label_kwargs = dict(labelType=\"text\", sele=\"@0\", showBackground=True, backgroundColor=\"black\")\n",
+    "    for ligand, affinity in zip(ligands, affinities):\n",
+    "        c = v.add_component(ligand)\n",
+    "        c.add_label(labelText=[str(affinity)], **label_kwargs)\n",
+    "    return v"
```

Comment:
Maybe use more descriptive variable names - v refers to viewer and c to canvas?

Reply:
`c` for component, but I agree, yes.